common:
    raw_lid_topic:  "/lslidar_point_cloud"
    lid_topic: "/point_cloud"
    imu_topic:  "/imu2"
    dyna_topic: "/dyna"
    chassis_topic: "/SocketCAN/Chassis_CAN_status"
    time_sync_en: false         # ONLY turn on when external time synchronization is really not possible
    
preprocess:
    lidar_type: 2                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR, 
    scan_line: 16
    scan_rate: 10                # only need to be set for velodyne, unit: Hz,
    blind: 5

mapping:
    acc_cov: 0.01
    gyr_cov: 0.01
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    enable_ground: true # true: enable the ground
    ground_cov_diag: [0.01,0.01,0.01]
    enable_dyna: true # true: enable the dyna
    dyna_cov_diag: [0.01,1000,1000,1000]
    dyna_time_delay: 0.0
    dyna_offset: 0.0
    vel_scale: 1.0 
    fov_degree:    360
    det_range:     50.0
    
    extrinsic_est_en:  true      # true: enable the online estimation of IMU-LiDAR extrinsic,2, 0, -0.4
    extrinsic_T: [  -0.15,0.55,1.0]
    extrinsic_R: [  -1, 0, 0,
                                0, -1, 0,
                                0, 0, 1   ]
    extrinsic_T_V: [-0.15,0.0, -0.67]
    extrinsic_R_V: [0, -1, 0,
                                    1, 0, 0,
                                    0, 0, 1]
publish:
    path_en:  true
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: true       # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: true  # true: output the point cloud scans in IMU-body-frame
    ground_publish_en: true # true: output the ground point cloud under vehicle
pcd_save:
    pcd_save_en: false
    interval: -1                 # how many LiDAR frames saved in each pcd file; 
                                 # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.